DAL/Repositories/Implementations/ProductRepository.cs

```diff
@@ -0,0 +1,59 @@
+﻿using DAL.Contexts;
+using DAL.Entities;
+using DAL.Repositories.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using System.Linq.Expressions;
+
+namespace DAL.Repositories.Implementations
+{
+    internal class ProductRepository : IProductRepository
+    {
+        private readonly EShopContext _context;
+
+        public ProductRepository(EShopContext context)
+        {
+            _context = context;
+        }
+
+        public void CreateAsync(Product product)
+        {
+            _context.Product.Add(product);
+        }
+
+        public void DeleteAsync(int id)
```

Comment:
pass the essence itself into the method like  public void DeleteAsync(Product product){_context.Product.Remove(product)};